server/command_import.go

```diff
@@ -0,0 +1,71 @@
+package main
+
+import (
+	"encoding/json"
+	"strings"
+
+	cloud "github.com/mattermost/mattermost-cloud/model"
+	"github.com/mattermost/mattermost-server/model"
+	"github.com/pkg/errors"
+)
+
+func (p *Plugin) runImportCommand(args []string, extra *model.CommandArgs) (*model.CommandResponse, bool, error) {
+	if len(args) == 0 {
+		return nil, true, errors.New("must provide an installation DNS")
+	}
+	installationDNS := standardizeName(args[0])
+
+	splitDNS := strings.Split(installationDNS, ".")
+	if len(splitDNS) < 2 {
+		return nil, true, errors.New("failed to parse DNS value")
+	}
+	name := splitDNS[0]
+
+	cloudInstall, err := p.cloudClient.GetInstallationByDNS(installationDNS, nil)
+	if err != nil {
+		return nil, false, errors.Wrap(err, "failed to get installation by DNS")
+	}
+	if cloudInstall == nil {
+		return nil, true, errors.New("no installation for the DNS provided")
+	}
+
+	installs, _, err := p.getInstallations()
+	if err != nil {
+		return nil, false, err
+	}
+
+	for _, install := range installs {
+		if install.ID == cloudInstall.ID {
+			return nil, true, errors.New("Installation has already been imported to cloud plugin")
+		}
+	}
+
+	if cloudInstall.OwnerID != extra.UserId {
+		cloudInstall.OwnerID = extra.UserId
```

Comment:
This means user A could import user B's installation, right? Don't think that's a problem just wanted to make sure that's expected

Reply:
Hey @jwilander, we actually made sure to cover this case with some logic above:

```
	for _, install := range installs {
		if install.ID == cloudInstall.ID {
			return nil, true, errors.New("Installation has already been imported to cloud plugin")
		}
	}
```

This should ensure the installation isn't already in the cloud plugin before we get here.

Reply:
Right, my mistake. I thought that was getting and only checking against the installations for the current user but it's all of them. Got it.